include/mgp.hpp

```diff
@@ -1814,10 +1815,12 @@ inline GraphRelationships::Iterator &GraphRelationships::Iterator::operator++()
   if (out_relationships_iterator_ != nullptr) {
     auto next = mgp::edges_iterator_next(out_relationships_iterator_);
 
-    if (next == nullptr) {
-      mgp::edges_iterator_destroy(out_relationships_iterator_);
-      out_relationships_iterator_ = nullptr;
+    if (next != nullptr) {
+      return *this;
     }
+
+    mgp::edges_iterator_destroy(out_relationships_iterator_);
+    out_relationships_iterator_ = nullptr;
```

Comment:
You don't return anything here.

Reply:
That’s the intended behavior!
If the iterator hasn’t found any more relationships on the current node, it doesn’t return anything. Instead, it moves on to go over the remaining nodes and return the first valid [outbound] relationship.

Reply:
That’s the intended behavior!
If the iterator hasn’t found any more relationships on the current node, it doesn’t return anything. Instead, it moves on to go over the remaining nodes and return the first valid [outbound] relationship.

Reply:
But that is a wrong behavior. Every non-void function **must** return a value. How would you check if something is returned or not? There is no way to achieve that. Therefore this function **must always** return a reference.

Reply:
Ahh sorry, I missed the scope of the function and thought you are never returrning from that function. My bad!